### DISCUSSION TITLE
Multiphase gas and Moving vitrual boundary surface

### DISCUSSION THREAD

**Discussion by SilberVogel** [InteractiveComputerGraphics/SPlisHSPlasH#154] [Q&A]
Hi all, I`m new in SPH and have some quistions about:

1.Does this software has possibilitys to set up and compute concentration of 4 or more phase gas with density near 1 kg/m3 and how to set up it?
2.Can I`m set up (and how to do it) virtual moving (with speed under M 0.9) boundary surface (that does not affect of itself geometry on main boundary) with massflow outlet (of this 4+ phase gas)?
3.In case of aerodynamics and compressible flow SPH (SPHlisHSPlasH and other) gives good results on validation or has some troubles?

Thx for attention!

**Comment by janbender** [comment]
1. I never tried this but if the density contrasts are not very high, this should work. 
2. Not completely sure what you want to do. However, you can have a moving boundary. You can take a look at the example scenes: AnimatedBody_2D.json (this requires a build with Python) or MotorScene.json (there is also a linear motor instead of an angular motor). 
3. The library implements an explicit and 2 implicit pressure solvers for compressible flow.

**Reply by SilberVogel** [reply]
> 1. I never tried this but if the density contrasts are not very high, this should work.
> 2. Not completely sure what you want to do. However, you can have a moving boundary. You can take a look at the example scenes: AnimatedBody_2D.json (this requires a build with Python) or MotorScene.json (there is also a linear motor instead of an angular motor).
> 3. The library implements an explicit and 2 implicit pressure solvers for compressible flow.

Many thanks for answer! 

2. I just want simulate aircraft jet engine exhaust at altitude... SPlisHSPlasH support energy quation? And different particle sizes?

Do you know some other codes for this task? (DualSPH, e.t.c)
